include/Scale.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Scale.h - holds information about a scale and its intervals
+ *
+ * Copyright (c) 2020 Martin Pavelek <he29.HS/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef SCALE_H
+#define SCALE_H
+
+#include <cmath>
+#include <cstdint>
+#include <vector>
+#include <QObject>
+#include <QString>
+
+#include "SerializingObject.h"
+
+class Interval : public SerializingObject
+{
+public:
+	Interval() : m_numerator(1), m_denominator(1), m_cents(0) {};
+	explicit Interval(float cents) : m_numerator(0), m_denominator(0), m_cents(cents) {};
+	Interval(uint32_t numerator, uint32_t denominator) :
+		m_numerator(numerator),
+		m_denominator(denominator > 0 ? denominator : 1),
+		m_cents(0) {};
+
+	float getRatio() const
+	{
+		if (m_denominator) {return static_cast<float>(m_numerator) / m_denominator;}
+		else {return powf(2.f, m_cents / 1200.f);}
```

Comment:
Good idea, thanks. I also wanted to learn how to use gprof, so this may be a nice little exercise.

Reply:
While changing this I also moved the constructors, since the now required initialization made them quite large. I tried to measure the difference, but it seems to be lost in the noise. (Looking only at export times; I wasn't able to use gprof since it requires LMMS to terminate correctly and on the machine I'm currently using it just becomes stuck waiting at some thread after shutdown, and killing it with SIGINT prevents gprof from saving the profiling data.)